be2-scala/src/main/scala/ch/epfl/pop/decentralized/HeartbeatGenerator.scala

```diff
@@ -0,0 +1,70 @@
+package ch.epfl.pop.decentralized
+
+import akka.actor.{Actor, ActorLogging, Props}
+import akka.pattern.AskableActorRef
+import ch.epfl.pop.model.network.method.Heartbeat
+import ch.epfl.pop.model.objects.{Channel, DbActorNAckException, Hash}
+import ch.epfl.pop.pubsub.AskPatternConstants
+import ch.epfl.pop.pubsub.graph.ErrorCodes
+import ch.epfl.pop.storage.DbActor
+import scala.util.{Failure, Success}
+import scala.collection.immutable.HashMap
+import scala.concurrent.Await
+
+// This actor will create heartbeat when asked to and send them to the actorRef it received
+case class HeartbeatGenerator(dbRef: AskableActorRef) extends Actor with ActorLogging with AskPatternConstants() {
+
+  /** retrieves the channels and their associated message ids
+    * @return
+    *   a map from the channels to message ids, None when errors are encountered
+    */
+  private def retrieveHeartbeatContent(): Option[HashMap[Channel, Set[Hash]]] = {
+    val askForChannels = dbRef ? DbActor.GetAllChannels()
```

Comment:
I added a pair of () to the receive method of the dbActor.scala, I hope the  bug is fixed for now. I am not sure to understand you suggestion. Certainly that moving heartbeat/getmessages_by_id logic will be moved to the db in coming pro, but do we have to wait until that to fix that bug ?

Reply:
The point was to avoid doing too much in the same pr, and I meant that you would do it the other way around: Removing the parenthesis in `HeartbeatGenerator`. As it broke the dbactor unit test, I took the liberty to fix the single missed parenthesis pair.